File: backend/src/main/java/com/NBE3_4_2_Team4/domain/member/bankAccount/repository/BankAccountRepository.java
package com.NBE3_4_2_Team4.domain.member.bankAccount.repository;

import com.NBE3_4_2_Team4.domain.member.member.entity.Member;
import com.NBE3_4_2_Team4.domain.member.bankAccount.entity.BankAccount;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface BankAccountRepository extends JpaRepository<BankAccount, Long> {

    List<BankAccount> findAllByMember(Member member);
}

File: backend/src/main/java/com/NBE3_4_2_Team4/domain/member/bankAccount/repository/BankAccountRepository.kt
package com.NBE3_4_2_Team4.domain.member.bankAccount.repository

import com.NBE3_4_2_Team4.domain.member.bankAccount.entity.BankAccount
import com.NBE3_4_2_Team4.domain.member.member.entity.Member
import org.springframework.data.jpa.repository.JpaRepository

interface BankAccountRepository : JpaRepository<BankAccount, Long> {

    fun findAllByMember(
        member: Member
    ) : List<BankAccount>

    fun existsByMemberIdAndBankCodeAndAccountNumberAndAccountHolder(
        memberId: Long,
        bankCode: String,
        accountNumber: String,
        accountHolder: String
    ) : Boolean
}